supervisor/snapshots/snapshot.py

```diff
@@ -370,17 +371,49 @@ def _folder_save(name):
             try:
                 _LOGGER.info("Snapshot folder %s", name)
                 with SecureTarFile(tar_name, "w", key=self._key) as tar_file:
-                    tar_file.add(
-                        origin_dir,
-                        arcname=".",
-                        filter=exclude_filter(MAP_FOLDER_EXCLUDE.get(name, [])),
+                    arcname = "."
+                    tar_file.add(origin_dir, arcname, False)
+                    add_directory_contents(
+                        tar_file, origin_dir, MAP_FOLDER_EXCLUDE.get(name, []), arcname
                     )
 
                 _LOGGER.info("Snapshot folder %s done", name)
                 self._data[ATTR_FOLDERS].append(name)
             except (tarfile.TarError, OSError) as err:
                 _LOGGER.warning("Can't snapshot folder %s: %s", name, err)
 
+        def add_directory_contents(
```

Comment:
Rename this function and export it to `utils/tar` and write tests for that function

Reply:
make sure we use this function also for add-on and other folder snapshots

Reply:
Just as mentioned [here](https://github.com/home-assistant/supervisor/pull/1795#discussion_r442111156):

I don't think that this should be part of the tar utils as it does not use any tar-related class/info or what benefit do we want to achieve?

Reply:
> make sure we use this function also for add-on and other folder snapshots

Can you give me a hint where this could be applied aswell? I only see folder handling `supervisor.snapshots.snapshot::store_folders`? Thus, all folder snapshots pass that method.

All other snapshots are taken from docker.